lib/govuk_component/button_helpers.rb

```diff
@@ -0,0 +1,101 @@
+# frozen_string_literal: true
+
+module GovukComponent
+  module ButtonHelpers
+    def govuk_button(content = nil, **tag_options, &block)
+      tag_options = prepend_classes('govuk-button', tag_options)
+      tag_options[:data] = { module: 'govuk-button', 'prevent-double-click': 'true' }
+      is_disabled(tag_options)
+
+      if block_given?
+        tag.button(tag_options, &block)
+      else
+        tag.button(content, tag_options)
+      end
+    end
+
+    def govuk_button_secondary(content = nil, **tag_options, &block)
+      tag_options = prepend_classes('govuk-button--secondary', tag_options)
+
+      govuk_button(content, tag_options, &block)
+    end
+
+    def govuk_button_warning(content = nil, **tag_options, &block)
+      tag_options = prepend_classes('govuk-button--warning', tag_options)
+
+      govuk_button(content, tag_options, &block)
+    end
+
+    def govuk_button_start(name = nil, options = nil, **tag_options, &block)
+      options = name if block_given?
+      url = url_for(options)
+      tag_options[:href] ||= url
+      tag_options = prepend_classes('govuk-button--start', tag_options)
+      govuk_link_button_options(tag_options)
+      content = block_given? ? capture(&block) : name || url
+
+      tag.a(tag_options) do
+        concat content
+        concat govuk_button_start_svg
+      end
+    end
+
+    def govuk_link_button(name = nil, options = nil, **tag_options, &block)
+      options = name if block_given?
+      url = url_for(options)
+      tag_options[:href] ||= url
+      govuk_link_button_options(tag_options)
+      is_disabled(tag_options)
+      content = block_given? ? capture(&block) : name || url
+
+      tag.a(content, tag_options, &block)
+    end
+
+    def govuk_link_button_secondary(name = nil, options = nil, **tag_options, &block)
+      tag_options = prepend_classes('govuk-button--secondary', tag_options)
+
+      govuk_link_button(name, options, tag_options, &block)
+    end
+
+    def govuk_link_button_warning(name = nil, options = nil, **tag_options, &block)
+      tag_options = prepend_classes('govuk-button--warning', tag_options)
+
+      govuk_link_button(name, options, tag_options, &block)
+    end
+
+    private
+
+    def is_disabled(options)
+      return unless options[:disabled].present? && options[:disabled].to_s == 'true'
+      options[:aria] = { disabled: 'true' }
+      prepend_classes('govuk-button--disabled', options)
+    end
+
+    def govuk_link_button_options(options)
+      options[:data] = { module: 'govuk-button' }
+      options[:draggable] = 'false'
+      options[:role] = 'button'
+      prepend_classes('govuk-button', options)
+    end
+
+    def govuk_button_start_svg
+      svg_options = {
+        class: 'govuk-button__start-icon',
+        xmlns: 'http://www.w3.org/2000/svg',
+        height: '19',
+        width: '17.5',
+        viewBox: '0 0 33 40',
+        'aria-hidden': 'true',
+        focusable: 'false'
+      }
+
+      svg_path_options = {
```

Comment:
i'd move this out to keep method size down (constant or method) gov_uk_button_start_svg_path_opts